### PR TITLE
Master

### PR DESCRIPTION
**Pull request by yunyuecoder**
错误处理bug 修复

### CODE REVIEW

**Comment by elbrujohalcon** [comment]
@yunyuecoder what's the purpose of this change?

**Comment by yunyuecoder** [comment]
when we get return {error, Reason},apns will not get the message ,so we need to resend message to apns,not wait to feedback

File: src/apns_connection.erl [hunk 1]
```diff
@@ -201,7 +201,11 @@ handle_cast(Msg, State) when is_record(Msg, apns_msg) ->
         ok ->
           {noreply, State#state{out_expires = Timeout}};
       {error, Reason} ->
-        apns_queue:fail(State#state.queue, Msg#apns_msg.id),
+        InfoLoggerFun = State#state.info_logger_fun,
+        InfoLoggerFun("[ ~p ]ssl error:", [Socket]),
+        ssl:close(Socket),
+        handle_cast(Msg, State#state{out_socket = undefined}),
```
Comment (elbrujohalcon) [inline src/apns_connection.erl]: You should check the response from `handle_cast` and update the `State` accordingly

**Comment by yunyuecoder** [comment]
and,we have a problem to tcp disconnect ,we cannot deal with TCP(SSL) disconnect,no any error return .

**Pull request closed without merging**